src/main/java/core/basesyntax/model/io/DataConverterImpl.java

```
@@ -0,0 +1,27 @@
+package core.basesyntax.model.io;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> lines) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+
+        for (String line : lines) {
+            String[] parts = line.split(",");
+
+            if (parts.length != 3) {
+                continue;
+            }
+
+            FruitTransaction.Operation op =
+                    FruitTransaction.Operation.fromCode(parts[0].trim());
```

The code relies on `FruitTransaction.Operation.fromCode(...)` which will throw IllegalArgumentException for invalid operation codes, but the converter currently does not catch this to provide a clearer message/context. Consider catching parsing-related exceptions (invalid operation code, NumberFormatException) and rethrowing a RuntimeException with an informative message mentioning the offending line. This aligns with the checklist guidance to not ignore exceptions and to rethrow informative RuntimeExceptions.